### PR TITLE
Support resource leeway parameter when simulating Soroban transactions.

### PR DESCRIPTION
**Pull request by overcat**
See #559

### CODE REVIEW

File: src/main/java/org/stellar/sdk/SorobanServer.java [hunk 1]
```diff
@@ -315,24 +319,41 @@ public GetLatestLedgerResponse getLatestLedger() throws IOException, SorobanRpcE
...
    * @return A {@link SimulateTransactionResponse} object containing the cost, footprint,
    *     result/auth requirements (if applicable), and error of the transaction.
    * @throws IOException If the request could not be executed due to cancellation, a connectivity
    *     problem or timeout. Because networks can fail during an exchange, it is possible that the
    *     remote server accepted the request before the failure.
    * @throws SorobanRpcErrorResponse If the Soroban-RPC instance returns an error response.
    */
-  public SimulateTransactionResponse simulateTransaction(Transaction transaction)
+  public SimulateTransactionResponse simulateTransaction(
+      Transaction transaction, @Nullable ResourceLeeway resourceLeeway)
```
Comment (sreuland) [inline src/main/java/org/stellar/sdk/SorobanServer.java:331]: is there a reason not to pass ResourceConfig here instead for more scalability? if more request attributes come up later, don't have to change this method signature again.

File: src/main/java/org/stellar/sdk/SorobanServer.java [hunk 2]
```diff
@@ -539,4 +560,12 @@ public enum Durability {
     TEMPORARY,
     PERSISTENT
   }
+
+  /** Describes additional resource leeways for transaction simulation. */
```
Comment (sreuland) [inline src/main/java/org/stellar/sdk/SorobanServer.java:564]: for class organization, can this class be defined statically in SimulateTransactionRequest instead?

File: src/test/java/org/stellar/sdk/SorobanServerTest.java [hunk 3]
```diff
@@ -774,6 +775,96 @@ public MockResponse dispatch(@NotNull RecordedRequest recordedRequest)
     mockWebServer.close();
   }
 
+  @Test
```
Comment (sreuland) [inline src/test/java/org/stellar/sdk/SorobanServerTest.java:778]: great coverage.

**Review by sreuland — APPROVED** [review]
I left a couple comments for consideration, otherwise lgtm!

File: src/main/java/org/stellar/sdk/SorobanServer.java (hunk 2 above)
Comment (overcat) [reply src/main/java/org/stellar/sdk/SorobanServer.java]: Of course, I have made the modifications.

File: src/main/java/org/stellar/sdk/SorobanServer.java (hunk 1 above)
Comment (overcat) [reply src/main/java/org/stellar/sdk/SorobanServer.java]: Of course, I have made the modifications.

**Pull request merged**